2008-08-11  Aurelien Jarno  <aurel32@debian.org>

	* sysdeps/mach/i386/machine-lock.h (__spin_unlock, __spin_try_lock):
	Fix asm constraints.

2008-08-02  Roland McGrath  <roland@frob.com>

	* sysdeps/mach/i386/machine-lock.h (__spin_unlock, __spin_try_lock):
	Add memory clobbers.

---
 sysdeps/mach/i386/machine-lock.h |    8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

--- a/sysdeps/mach/i386/machine-lock.h
+++ b/sysdeps/mach/i386/machine-lock.h
@@ -1,5 +1,5 @@
 /* Machine-specific definition for spin locks.  i386 version.
-   Copyright (C) 1994, 1997, 2007 Free Software Foundation, Inc.
+   Copyright (C) 1994,1997,2007,2008 Free Software Foundation, Inc.
    This file is part of the GNU C Library.
 
    The GNU C Library is free software; you can redistribute it and/or
@@ -43,7 +43,8 @@
 {
    register int __unlocked;
    __asm__ __volatile ("xchgl %0, %1"
-		       : "=&r" (__unlocked), "=m" (*__lock) : "0" (0));
+		       : "=&r" (__unlocked), "=m" (*__lock) : "0" (0)
+		       : "memory");
 }
 #endif
 
@@ -57,7 +58,8 @@
 {
   register int __locked;
   __asm__ __volatile ("xchgl %0, %1"
-		      : "=&r" (__locked), "=m" (*__lock) : "0" (1));
+		      : "=&r" (__locked), "=m" (*__lock) : "0" (1)
+		      : "memory");
   return !__locked;
 }
 #endif
